lib/python/picongpu/plugins/data/compare.py

```diff
@@ -0,0 +1,222 @@
+        iteration : (unsigned) int [unitless]
+            The iteration at which to read the data.
+            A list of iterations is allowed as well.
+            ``None`` refers to the list of all available iterations.
+
+        Returns
+        -------params,values,other_params
+        variation : np.array
+            values of the variated input parameter
+        observables : np.array
+            calculated observed values for the respectiv input parameter
+        other_params: dict
+            values of other variated input parameters
+        """
+        # get list of path to files
+        if iteration is not None:
+            if not isinstance(iteration, collections.Iterable):
+                iteration = np.array([iteration])
+        sims_rundir, params = self.get_data_path(species, variation,
+                                                 observable,
+                                                 species_filter)
+        values = []
+        params = []
+        other_params = []
+        for sim in (sims_rundir):
+            sim_json = os.path.join(os.path.split(sim)[0], 'params.json')
+            if not os.path.isfile(sim_json):
+                raise IOError('The file {} does not exist.\n'
+                              'Did the simulation already run?'
+                              .format(self.path_to_json))
+            other_param = dict()
+            with open(sim_json) as f:
+                range_dict = json.load(f)
+                param = range_dict[variation]['values'][0]
+                for key in self.fix_param_dict.keys():
+                    other_param[key] = '{:0.1e}'.format(
+                                       range_dict[key]['values'][0], 2)
+            try:
+                if observable is not "emittance":
+                    counts, bins = EnergyHistogramData(sim).get(
+                                   species, species_filter, iteration)
+                    start = 80
+                    limitE = 500
+                    data = counts[start:]
+                    for i in np.arange(data.size):
+                        if (max(data) == data[i]):
+                            peakEnergy = i
+                    for j in np.arange(data.size-peakEnergy):
+                        if (max(data)/3 - data[peakEnergy + j] > 0.0):
+                            FWHM_max = ((peakEnergy + j + start) *
+                                        limitE / len(bins))
+                            break
+                    for k in np.arange(peakEnergy):
+                        if (max(data)/3-data[peakEnergy - k] > 0.0):
+                            FWHM_min = ((peakEnergy - k + start)
+                                        * limitE / len(bins))
+                            break
+                if observable == "bunchEnergy":
+                    maxEnergy = 0.5 * (FWHM_max + FWHM_min)
+                    value = maxEnergy  # unit: MeV
+                if observable == "energySpread":
```

Comment:
Shouldn't that rather be a big `if ... else if ...else if...else` block instead?